xlators/cluster/dht/src/dht-rebalance.c

```diff
@@ -3221,12 +3217,16 @@ int static gf_defrag_get_entry(xlator_t *this, int i,
     }
 
 out:
-    if (ret == 0) {
-        *container = tmp_container;
-    } else {
-        if (tmp_container) {
-            gf_defrag_free_container(tmp_container);
+    if (defrag->defrag_status == GF_DEFRAG_STATUS_STARTED) {
+        if (ret == 0) {
+            *container = tmp_container;
+        } else {
+            if (tmp_container) {
+                gf_defrag_free_container(tmp_container);
+            }
         }
+    } else {
+        gf_defrag_free_container(tmp_container);
```

Comment:
nit: add a a null check to tmp_container like l3224

Reply:
Actually, looking into the implementation of gf_defrag_free_container, it already performs null ptr check, so there's no need for the check in line 3224. Will push a revision.